platform/event/event_payload.proto

```diff
@@ -0,0 +1,64 @@
+/**
+ * # Event Payload
+ * A payload for an event intended to be transferred between nodes.
+ * This message forms the base unit of data communicated for consensus.
+ *
+ * ### Keywords
+ * The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+ * "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+ * document are to be interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119).
+ */
+syntax = "proto3";
+
+package proto;
+
+/*
+ * Hedera Network Services Protobuf
+ *
+ * Copyright (C) 2018 - 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import "state_signature_payload.proto";
+
+option java_package = "com.swirlds.proto.legacy.event";
+// <<<pbj.java_package = "com.swirlds.proto.event">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+/**
+ * An Event Payload gossiped between nodes as part of events.<br/>
+ * <br/>
+ * Each node MUST extract this payload and process according to the type
+ * of payload encoded.<br/>
+ * Both the platform and the application built on that platform MAY define event
+ * payloads.<br/>
+ * The encoded data SHOULD be a serialized protobuf message.<br/>
+ */
+message EventPayload {
+  oneof payload {
+      /**
+       * An application payload.<br/>
+       * The contents of this payload SHALL be defined by the application
+       * subsystem that created the event.<br/>
+       * The contents SHOULD be a serialized protobuf message.
+       */
+      bytes application_payload = 1;
+      /**
+       * A state signature.<br/>
+       * This payload SHOULD be a valid state signature for a state snapshot.
```

Comment:
The mild "should" is perhaps a bit weak here.
Is this message going to be used with some other content here?
If not, then `SHALL` is probably the more precise choice.

This text needs to form a specification sufficient that another, unrelated, team can correctly implement the communication protocol these messages define and support. While perhaps not strictly necessary for this PR; ideally we'll continuously improve the quality of these specification documents.

Reply:
changed